<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Gene Ontology • simona</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Gene Ontology">
<meta property="og:description" content="simona">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">simona</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/v1_dag.html">ontology_DAG: a class for ontologies</a>
    </li>
    <li>
      <a href="../articles/v2_GO.html">Gene Ontology</a>
    </li>
    <li>
      <a href="../articles/v3_import.html">Import ontology files</a>
    </li>
    <li>
      <a href="../articles/v4_information_content.html">Information content</a>
    </li>
    <li>
      <a href="../articles/v5_term_similarity.html">Term similarity</a>
    </li>
    <li>
      <a href="../articles/v6_group_similarity.html">Similarity between two groups of terms</a>
    </li>
    <li>
      <a href="../articles/v7_dag_visualization.html">Visualize DAGs</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/jokergoo/simone/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Gene Ontology</h1>
                        <h4 data-toc-skip class="author">Zuguang Gu ( <a href="mailto:z.gu@dkfz.de" class="email">z.gu@dkfz.de</a> )</h4>
            
            <h4 data-toc-skip class="date">2023-08-04</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/jokergoo/simone/blob/HEAD/vignettes/v2_GO.Rmd" class="external-link"><code>vignettes/v2_GO.Rmd</code></a></small>
      <div class="hidden name"><code>v2_GO.Rmd</code></div>

    </div>

    
    
<p>Gene Ontology is the most widely used bio-ontologies. On Bioconductor, there are standard packages for GO (<strong>GO.db</strong>) and organism-specific GO annotations (<strong>org.*.db</strong>). In <strong>simona</strong>, there is a helper function <code><a href="../reference/create_ontology_DAG_from_GO_db.html">create_ontology_DAG_from_GO_db()</a></code> which makes use of the Biocoductor standard GO packages and constructs a DAG object automatically.</p>
<div class="section level2">
<h2 id="create-the-go-dag-object">Create the GO DAG object<a class="anchor" aria-label="anchor" href="#create-the-go-dag-object"></a>
</h2>
<p>GO has three namespaces (or ontologies): biological process (BP), Molecular function (MF) and celullar component (CC). The three GO namespaces are mutually exclusive, so the first argument of <code><a href="../reference/create_ontology_DAG_from_GO_db.html">create_ontology_DAG_from_GO_db()</a></code> is the GO namespace.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/simone" class="external-link">simona</a></span><span class="op">)</span></span>
<span><span class="va">dag</span> <span class="op">=</span> <span class="fu"><a href="../reference/create_ontology_DAG_from_GO_db.html">create_ontology_DAG_from_GO_db</a></span><span class="op">(</span><span class="st">"BP"</span><span class="op">)</span></span>
<span><span class="va">dag</span></span></code></pre></div>
<pre><code><span><span class="co">## An ontology_DAG object:</span></span>
<span><span class="co">##   Source: GO BP / GO.db package </span></span>
<span><span class="co">##   28140 terms / 56449 relations</span></span>
<span><span class="co">##   Root: GO:0008150 </span></span>
<span><span class="co">##   Terms: GO:0000001, GO:0000002, GO:0000003, GO:0000011, ...</span></span>
<span><span class="co">##   Max depth: 18 </span></span>
<span><span class="co">##   Aspect ratio: 342.43:1 (based on the longest distance to root)</span></span>
<span><span class="co">##                 780.22:1 (based on the shortest distance to root)</span></span>
<span><span class="co">##   Relations: is_a, part_of</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## With the following columns in the metadata data frame:</span></span>
<span><span class="co">##   id, name, definition</span></span></code></pre>
<p>There are three main GO relations: “is_a”, “part_of” and “regulates”. “regulates” has two child relation types in GO: “negatively_regulates” and “positively_regulates”. So when “regulates” is selected, the two child relation types are automatically selected. By default only “is_a” and “part_of” are selected.</p>
<p>You can set a subset of relation types with the argument <code>relations</code>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/create_ontology_DAG_from_GO_db.html">create_ontology_DAG_from_GO_db</a></span><span class="op">(</span><span class="st">"BP"</span>, relations <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"part of"</span>, <span class="st">"regulates"</span><span class="op">)</span><span class="op">)</span>  <span class="co"># "part_of" is also OK</span></span></code></pre></div>
<pre><code><span><span class="co">## An ontology_DAG object:</span></span>
<span><span class="co">##   Source: GO BP / GO.db package </span></span>
<span><span class="co">##   28140 terms / 65107 relations</span></span>
<span><span class="co">##   Root: GO:0008150 </span></span>
<span><span class="co">##   Terms: GO:0000001, GO:0000002, GO:0000003, GO:0000011, ...</span></span>
<span><span class="co">##   Max depth: 19 </span></span>
<span><span class="co">##   Aspect ratio: 283.07:1 (based on the longest distance to root)</span></span>
<span><span class="co">##                 1011.88:1 (based on the shortest distance to root)</span></span>
<span><span class="co">##   Relations: is_a, negatively_regulates, part_of, positively_regulates,</span></span>
<span><span class="co">##              regulates</span></span>
<span><span class="co">##   Relation types may have hierarchical relations.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## With the following columns in the metadata data frame:</span></span>
<span><span class="co">##   id, name, definition</span></span></code></pre>
<p>“is_a” is always selected because this is primary semantic relation type. So if you only want to include “is_a” relation, you can assign an empty vector to <code>relations</code>:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/create_ontology_DAG_from_GO_db.html">create_ontology_DAG_from_GO_db</a></span><span class="op">(</span><span class="st">"BP"</span>, relations <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">character</a></span><span class="op">(</span><span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="co"># or NULL, NA</span></span></code></pre></div>
<p>Or you can apply <code><a href="../reference/dag_filter.html">dag_filter()</a></code> after DAG is generated.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dag</span> <span class="op">=</span> <span class="fu"><a href="../reference/create_ontology_DAG_from_GO_db.html">create_ontology_DAG_from_GO_db</a></span><span class="op">(</span><span class="st">"BP"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/dag_filter.html">dag_filter</a></span><span class="op">(</span><span class="va">dag</span>, relations <span class="op">=</span> <span class="st">"is_a"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="add-gene-annotation">Add gene annotation<a class="anchor" aria-label="anchor" href="#add-gene-annotation"></a>
</h2>
<p>Gene annotation can be set with the argument <code>org_db</code>. The value is an <code>OrgDb</code> object of the corresponding organism.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></span>
<span><span class="va">dag</span> <span class="op">=</span> <span class="fu"><a href="../reference/create_ontology_DAG_from_GO_db.html">create_ontology_DAG_from_GO_db</a></span><span class="op">(</span><span class="st">"BP"</span>, org_db <span class="op">=</span> <span class="va">org.Hs.eg.db</span><span class="op">)</span></span>
<span><span class="va">dag</span></span></code></pre></div>
<pre><code><span><span class="co">## An ontology_DAG object:</span></span>
<span><span class="co">##   Source: GO BP / GO.db package </span></span>
<span><span class="co">##   28140 terms / 56449 relations</span></span>
<span><span class="co">##   Root: GO:0008150 </span></span>
<span><span class="co">##   Terms: GO:0000001, GO:0000002, GO:0000003, GO:0000011, ...</span></span>
<span><span class="co">##   Max depth: 18 </span></span>
<span><span class="co">##   Aspect ratio: 342.43:1 (based on the longest distance to root)</span></span>
<span><span class="co">##                 780.22:1 (based on the shortest distance to root)</span></span>
<span><span class="co">##   Relations: is_a, part_of</span></span>
<span><span class="co">##   Annotations are available.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## With the following columns in the metadata data frame:</span></span>
<span><span class="co">##   id, name, definition</span></span></code></pre>
<p>For standard organism packages on Biocoductor, the <code>OrgDb</code> object always has the same name as the package, so the name of the organism package can also be set to <code>org_db</code>:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/create_ontology_DAG_from_GO_db.html">create_ontology_DAG_from_GO_db</a></span><span class="op">(</span><span class="st">"BP"</span>, org_db <span class="op">=</span> <span class="st">"org.Hs.eg.db"</span><span class="op">)</span></span></code></pre></div>
<p>Similarly, if the analysis is applied on mouse, the mouse organism package can be set to <code>org_db</code>. If the mouse organism package is not installed yet, it will be installed automatically.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/create_ontology_DAG_from_GO_db.html">create_ontology_DAG_from_GO_db</a></span><span class="op">(</span><span class="st">"BP"</span>, org_db <span class="op">=</span> <span class="st">"org.Mm.eg.db"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="meta-data-frame">Meta data frame<a class="anchor" aria-label="anchor" href="#meta-data-frame"></a>
</h2>
<p>There are additional meta columns attached to the DAG object. They can be accessed by <code>mcols()</code>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">dag</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                    id                             name</span></span>
<span><span class="co">## GO:0000001 GO:0000001        mitochondrion inheritance</span></span>
<span><span class="co">## GO:0000002 GO:0000002 mitochondrial genome maintenance</span></span>
<span><span class="co">## GO:0000003 GO:0000003                     reproduction</span></span>
<span><span class="co">## GO:0000011 GO:0000011              vacuole inheritance</span></span>
<span><span class="co">## GO:0000012 GO:0000012       single strand break repair</span></span>
<span><span class="co">## GO:0000017 GO:0000017        alpha-glucoside transport</span></span>
<span><span class="co">##                                                                                                                                                                                                                                                                                                   definition</span></span>
<span><span class="co">## GO:0000001                                                                                                           The distribution of mitochondria, including the mitochondrial genome, into daughter cells after mitosis or meiosis, mediated by interactions between mitochondria and the cytoskeleton.</span></span>
<span><span class="co">## GO:0000002                                                                                                                                                 The maintenance of the structure and integrity of the mitochondrial genome; includes replication and segregation of the mitochondrial chromosome.</span></span>
<span><span class="co">## GO:0000003                                                                                                                                                                      The production of new individuals that contain some portion of genetic material inherited from one or more parent organisms.</span></span>
<span><span class="co">## GO:0000011                                                                                                                                                        The distribution of vacuoles into daughter cells after mitosis or meiosis, mediated by interactions between vacuoles and the cytoskeleton.</span></span>
<span><span class="co">## GO:0000012                                                                                                                                                      The repair of single strand breaks in DNA. Repair of such breaks is mediated by the same enzyme systems as are used in base excision repair.</span></span>
<span><span class="co">## GO:0000017 The directed movement of alpha-glucosides into, out of or within a cell, or between cells, by means of some agent such as a transporter or pore. Alpha-glucosides are glycosides in which the sugar group is a glucose residue, and the anomeric carbon of the bond is in an alpha configuration.</span></span></code></pre>
<p>The additional information of GO terms is from the <strong>GO.db</strong> package. The row order of the meta data frame is the same as in <code>dag_all_terms(dag)</code>.</p>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.2.0 (2022-04-22)</span></span>
<span><span class="co">## Platform: x86_64-apple-darwin17.0 (64-bit)</span></span>
<span><span class="co">## Running under: macOS Big Sur/Monterey 10.16</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib</span></span>
<span><span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] C/UTF-8/C/C/C/C</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] org.Hs.eg.db_3.16.0  AnnotationDbi_1.60.2 IRanges_2.32.0      </span></span>
<span><span class="co">## [4] S4Vectors_0.36.2     Biobase_2.58.0       BiocGenerics_0.44.0 </span></span>
<span><span class="co">## [7] simona_0.99.0        knitr_1.43          </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] Rcpp_1.0.11            circlize_0.4.16        GO.db_3.16.0          </span></span>
<span><span class="co">##  [4] Biostrings_2.66.0      png_0.1-8              rprojroot_2.0.3       </span></span>
<span><span class="co">##  [7] digest_0.6.33          foreach_1.5.2          GenomeInfoDb_1.34.9   </span></span>
<span><span class="co">## [10] R6_2.5.1               Polychrome_1.5.1       RSQLite_2.3.1         </span></span>
<span><span class="co">## [13] evaluate_0.21          httr_1.4.6             zlibbioc_1.44.0       </span></span>
<span><span class="co">## [16] GlobalOptions_0.1.2    rlang_1.1.1            jquerylib_0.1.4       </span></span>
<span><span class="co">## [19] blob_1.2.4             GetoptLong_1.1.0       rmarkdown_2.23        </span></span>
<span><span class="co">## [22] pkgdown_2.0.7          textshaping_0.3.6      desc_1.4.2            </span></span>
<span><span class="co">## [25] stringr_1.5.0          RCurl_1.98-1.12        igraph_1.5.0          </span></span>
<span><span class="co">## [28] bit_4.0.5              compiler_4.2.0         xfun_0.39             </span></span>
<span><span class="co">## [31] pkgconfig_2.0.3        systemfonts_1.0.4      shape_1.4.6           </span></span>
<span><span class="co">## [34] htmltools_0.5.5        KEGGREST_1.38.0        GenomeInfoDbData_1.2.9</span></span>
<span><span class="co">## [37] codetools_0.2-19       matrixStats_1.0.0      crayon_1.5.2          </span></span>
<span><span class="co">## [40] bitops_1.0-7           grid_4.2.0             jsonlite_1.8.7        </span></span>
<span><span class="co">## [43] lifecycle_1.0.3        DBI_1.1.3              magrittr_2.0.3        </span></span>
<span><span class="co">## [46] cli_3.6.1              stringi_1.7.12         cachem_1.0.8          </span></span>
<span><span class="co">## [49] XVector_0.38.0         fs_1.6.2               scatterplot3d_0.3-44  </span></span>
<span><span class="co">## [52] doParallel_1.0.17      xml2_1.3.5             bslib_0.5.0           </span></span>
<span><span class="co">## [55] ragg_1.2.5             vctrs_0.6.3            rjson_0.2.21          </span></span>
<span><span class="co">## [58] RColorBrewer_1.1-3     iterators_1.0.14       tools_4.2.0           </span></span>
<span><span class="co">## [61] bit64_4.0.5            glue_1.6.2             purrr_1.0.1           </span></span>
<span><span class="co">## [64] parallel_4.2.0         fastmap_1.1.1          yaml_2.3.7            </span></span>
<span><span class="co">## [67] clue_0.3-64            colorspace_2.1-0       cluster_2.1.4         </span></span>
<span><span class="co">## [70] TSP_1.2-4              ComplexHeatmap_2.15.4  memoise_2.0.1         </span></span>
<span><span class="co">## [73] sass_0.4.7</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Zuguang Gu.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
