<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Information content • simona</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Information content">
<meta property="og:description" content="simona">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">simona</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/v1_dag.html">ontology_DAG: a class for ontologies</a>
    </li>
    <li>
      <a href="../articles/v2_GO.html">Gene Ontology</a>
    </li>
    <li>
      <a href="../articles/v3_import.html">Import ontology files</a>
    </li>
    <li>
      <a href="../articles/v4_information_content.html">Information content</a>
    </li>
    <li>
      <a href="../articles/v5_term_similarity.html">Term similarity</a>
    </li>
    <li>
      <a href="../articles/v6_group_similarity.html">Similarity between two groups of terms</a>
    </li>
    <li>
      <a href="../articles/v7_dag_visualization.html">Visualize DAGs</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/jokergoo/simone/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Information content</h1>
                        <h4 data-toc-skip class="author">Zuguang Gu ( <a href="mailto:z.gu@dkfz.de" class="email">z.gu@dkfz.de</a> )</h4>
            
            <h4 data-toc-skip class="date">2023-08-04</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/jokergoo/simone/blob/HEAD/vignettes/v4_information_content.Rmd" class="external-link"><code>vignettes/v4_information_content.Rmd</code></a></small>
      <div class="hidden name"><code>v4_information_content.Rmd</code></div>

    </div>

    
    
<p>The methods of information content implemented in <strong>simona</strong> are mainly from the <a href="https://oup.silverchair-cdn.com/oup/backfile/Content_public/Journal/bib/18/5/10.1093_bib_bbw067/2/bbw067_supplementary_file.pdf?Expires=1692176766&amp;Signature=Ks4C96x-kinySVP7rzVzMn8cGnQWXOqkB-6gYHhs48Bl9A2Vj~-mcI9pEnvntNBvjBD1hp4ZncMWUNT-y1ArrrQgDJ0dCCEJp5LG0Ag2oPWsC5NFEeUQiYNZCXal8Kwa3Ze1vtDHjIwZpT9-sZhn9ryqKHd3C88RvuGx10jcSYRL1Pec0F~a9CnrbivieCfFELg2OQTf5zpAobxhlqzWqv~sa1b62PunNZ~XkmnTGOJoSMQFnNaBy~bDNEf9Pd6eRYaOc8MUh5Br4xB4y8mivT6Et2vKzLrGJFQjYpvouQpBu0nf03xwHVmNFN5-6LsDnpyQPCy4nQ5DHYltVHkEOA__&amp;Key-Pair-Id=APKAIE5G5CRDK6RD3PGA" class="external-link">supplementary file</a> of the paper <a href="https://doi.org/10.1093/bib/bbw067" class="external-link">“Mazandu et al., Gene Ontology semantic similarity tools: survey on features and challenges for biological knowledge discovery. Briefings in Bioinformatics 2017”</a>. Original denotations have been slightly modified to make them more consistent. Also more explanations have been added in this vignette. Credits go to the original paper.</p>
<div class="section level2">
<h2 id="denotations">Denotations<a class="anchor" aria-label="anchor" href="#denotations"></a>
</h2>
<p>The following denotations will be used throughout the vignette.</p>
<table class="table">
<colgroup>
<col width="19%">
<col width="80%">
</colgroup>
<thead><tr class="header">
<th align="left">Denotation</th>
<th align="left">Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left"><span class="math inline">\(r\)</span></td>
<td align="left">The root term of the DAG.</td>
</tr>
<tr class="even">
<td align="left"><span class="math inline">\(\delta(x)\)</span></td>
<td align="left">The depth of a term <span class="math inline">\(x\)</span> in the DAG, which is the longest distance from root <span class="math inline">\(r\)</span>.</td>
</tr>
<tr class="odd">
<td align="left"><span class="math inline">\(\delta_s(x)\)</span></td>
<td align="left">The length of the longest path from root <span class="math inline">\(r\)</span> to a term <span class="math inline">\(x\)</span> via term <span class="math inline">\(s\)</span>.</td>
</tr>
<tr class="even">
<td align="left"><span class="math inline">\(\delta_\max\)</span></td>
<td align="left">The maximal depth in the DAG.</td>
</tr>
<tr class="odd">
<td align="left"><span class="math inline">\(\eta(x)\)</span></td>
<td align="left">The height of term <span class="math inline">\(x\)</span> in the DAG, which is the longest finite distance to leaf terms.</td>
</tr>
<tr class="even">
<td align="left"><span class="math inline">\(\mathcal{C}_s\)</span></td>
<td align="left">The set of child terms of term <span class="math inline">\(s\)</span>.</td>
</tr>
<tr class="odd">
<td align="left"><span class="math inline">\(\mathcal{P}_s\)</span></td>
<td align="left">The set of parents terms of term <span class="math inline">\(s\)</span>.</td>
</tr>
<tr class="even">
<td align="left"><span class="math inline">\(\mathcal{A}_s\)</span></td>
<td align="left">The set of ancestor terms of term <span class="math inline">\(s\)</span>.</td>
</tr>
<tr class="odd">
<td align="left"><span class="math inline">\(\mathcal{A}_s^+\)</span></td>
<td align="left">The set of ancestor terms of term <span class="math inline">\(s\)</span>, including <span class="math inline">\(s\)</span> itself.</td>
</tr>
<tr class="even">
<td align="left"><span class="math inline">\(\mathcal{D}_s\)</span></td>
<td align="left">The set of offspring terms of term <span class="math inline">\(s\)</span>.</td>
</tr>
<tr class="odd">
<td align="left"><span class="math inline">\(\mathcal{D}_s^+\)</span></td>
<td align="left">The set of offspring terms of term <span class="math inline">\(s\)</span>, including <span class="math inline">\(s\)</span> itself.</td>
</tr>
<tr class="even">
<td align="left"><span class="math inline">\(\mathcal{L}_s\)</span></td>
<td align="left">The set of leaf terms that term <span class="math inline">\(s\)</span> can reach.</td>
</tr>
<tr class="odd">
<td align="left"><span class="math inline">\(\left| A \right|\)</span></td>
<td align="left">Number of elements in set <span class="math inline">\(A\)</span>.</td>
</tr>
<tr class="even">
<td align="left"><span class="math inline">\(D_\mathrm{sp}(a, b)\)</span></td>
<td align="left">The shortest distance bewteen <span class="math inline">\(a\)</span> and <span class="math inline">\(b\)</span>.</td>
</tr>
<tr class="odd">
<td align="left"><span class="math inline">\(\mathrm{len}(a, b)\)</span></td>
<td align="left">The longest distance bewteen <span class="math inline">\(a\)</span> and <span class="math inline">\(b\)</span>.</td>
</tr>
<tr class="even">
<td align="left"><span class="math inline">\(\mathrm{len}_s(a, b)\)</span></td>
<td align="left">The length of the longest path from <span class="math inline">\(a\)</span> and <span class="math inline">\(b\)</span> via <span class="math inline">\(s\)</span>.</td>
</tr>
<tr class="odd">
<td align="left"><span class="math inline">\(\mathrm{CA}(a, b)\)</span></td>
<td align="left">The set of common ancestors of term <span class="math inline">\(a\)</span> and <span class="math inline">\(b\)</span>, i.e. <span class="math inline">\(\mathrm{CA}(a, b) = \mathcal{A}_a^+ \cap \mathcal{A}_b^+\)</span>
</td>
</tr>
<tr class="even">
<td align="left"><span class="math inline">\(\mathrm{LCA}(a, b)\)</span></td>
<td align="left">Lowest common ancestor of <span class="math inline">\(a\)</span> and <span class="math inline">\(b\)</span>, which is the common ancestor with the largest depth in DAG, i.e. <span class="math display">\[\operatorname*{argmax}_{t \in \mathrm{CA}(a, b)} \delta(t)\]</span> There might be more than one LCA terms for given two terms, to simplify the calculation, the one with the smallest distance to <span class="math inline">\(a\)</span> and <span class="math inline">\(b\)</span> is used.</td>
</tr>
<tr class="odd">
<td align="left"><span class="math inline">\(\mathrm{NCA}(a, b)\)</span></td>
<td align="left">Nearest common ancestor of <span class="math inline">\(a\)</span> and <span class="math inline">\(b\)</span>, i.e. <span class="math display">\[\operatorname*{argmin}_{t \in \mathrm{CA}(a, b)} \left( D_\mathrm{sp}(t, a) + D_\mathrm{sp}(t, b) \right)\]</span> If there are more than one NCA terms, the one with the highest depth is used.</td>
</tr>
<tr class="even">
<td align="left"><span class="math inline">\(\mathrm{MICA}(a, b)\)</span></td>
<td align="left">Most informative common ancestor of <span class="math inline">\(a\)</span> and <span class="math inline">\(b\)</span>, i.e. <span class="math display">\[\operatorname*{argmax}_{t \in \mathrm{CA}(a, b)} \left( \mathrm{IC}(t) \right )\]</span> There might be more than one MICA terms for given two terms, the one with the smallest distance to <span class="math inline">\(a\)</span> and <span class="math inline">\(b\)</span> is used.</td>
</tr>
<tr class="odd">
<td align="left"><span class="math inline">\(G_s\)</span></td>
<td align="left">The set of annotated items on term <span class="math inline">\(s\)</span>.</td>
</tr>
</tbody>
</table>
<p>Assume term <span class="math inline">\(a\)</span> is an ancestor of term <span class="math inline">\(b\)</span>, <span class="math inline">\(D_\mathrm{sp}(a, b)\)</span> (the order of <span class="math inline">\(a\)</span> and <span class="math inline">\(b\)</span> does not matter) is the normal shortest distance from <span class="math inline">\(a\)</span> to <span class="math inline">\(b\)</span> in a directed graph. The definition is similar for <span class="math inline">\(\mathrm{len}(a, b)\)</span>.</p>
<p>If term <span class="math inline">\(a\)</span> and <span class="math inline">\(b\)</span> are not in offspring/ancestor relationship, i.e. <span class="math inline">\(a\)</span> is not an ancestor of <span class="math inline">\(b\)</span>, and <span class="math inline">\(b\)</span> is not an ancestor of <span class="math inline">\(a\)</span>, then</p>
<p><span class="math display">\[
\begin{align*}
D_\mathrm{sp}(a, b) &amp;= \min_{t \in \mathrm{CA}(a, b)} \left( D_\mathrm{sp}(t, a) + D_\mathrm{sp}(t, b) \right) \\
\mathrm{len}(a, b) &amp;= \max_{t \in \mathrm{CA}(a, b)} \left( \mathrm{len}(t, a) + \mathrm{len}(t, b) \right)
\end{align*}
\]</span></p>
<p>For many information content (IC) definitions, it has the following form:</p>
<p><span class="math display">\[ \mathrm{IC}(x) = -\log(p(x)) \]</span></p>
<p>where <span class="math inline">\(p(x)\)</span> is the probability of term <span class="math inline">\(x\)</span> under a certain measure in the DAG. If a term has a smaller probability, it means it is more specific, the form <span class="math inline">\(-\log(p(x))\)</span> will have a larger value, thus it is called the “information content”. Basically the more to the leaves in the DAG, a smaller p-value is gained and the higher information content a term has.</p>
<p>In the following sections, I will use Gene Ontology as an example to demonstrate various IC methods supported in the <strong>simona</strong> package. Note here I also set the gene annotation because some IC methods depend on the number of annotated genes on GO terms.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/simone" class="external-link">simona</a></span><span class="op">)</span></span>
<span><span class="va">dag</span> <span class="op">=</span> <span class="fu"><a href="../reference/create_ontology_DAG_from_GO_db.html">create_ontology_DAG_from_GO_db</a></span><span class="op">(</span>namespace <span class="op">=</span> <span class="st">"BP"</span>, org_db <span class="op">=</span> <span class="st">"org.Hs.eg.db"</span><span class="op">)</span></span>
<span><span class="va">dag</span></span></code></pre></div>
<pre><code><span><span class="co">## An ontology_DAG object:</span></span>
<span><span class="co">##   Source: GO BP / GO.db package </span></span>
<span><span class="co">##   28140 terms / 56449 relations</span></span>
<span><span class="co">##   Root: GO:0008150 </span></span>
<span><span class="co">##   Terms: GO:0000001, GO:0000002, GO:0000003, GO:0000011, ...</span></span>
<span><span class="co">##   Max depth: 18 </span></span>
<span><span class="co">##   Aspect ratio: 342.43:1 (based on the longest distance to root)</span></span>
<span><span class="co">##                 780.22:1 (based on the shortest distance to root)</span></span>
<span><span class="co">##   Relations: is_a, part_of</span></span>
<span><span class="co">##   Annotations are available.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## With the following columns in the metadata data frame:</span></span>
<span><span class="co">##   id, name, definition</span></span></code></pre>
<p>The wrapper function <code><a href="../reference/term_IC.html">term_IC()</a></code> calculates information contents of terms in the DAG with a specific method. Note the method name can be partially matched.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/term_IC.html">term_IC</a></span><span class="op">(</span><span class="va">dag</span>, method <span class="op">=</span> <span class="va">...</span>, control <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">...</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="ic_offspring">IC_offspring<a class="anchor" aria-label="anchor" href="#ic_offspring"></a>
</h2>
<p>The information content is calculated as:</p>
<p><span class="math display">\[ \mathrm{IC}(x) = -\log \left(  \frac{|\mathcal{D}_x^+|}{|\mathcal{D}_r^+|} \right) \]</span></p>
<p>In the bracket is the fraction of number of offspring terms to the total number of terms in the DAG.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/term_IC.html">term_IC</a></span><span class="op">(</span><span class="va">dag</span>, method <span class="op">=</span> <span class="st">"IC_offspring"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="ic_height">IC_height<a class="anchor" aria-label="anchor" href="#ic_height"></a>
</h2>
<p>It measures the relative position of a term in the DAG:</p>
<p><span class="math display">\[ \mathrm{IC}(x) = -\log \left( \frac{\eta(x) + 1}{\delta(x) + \eta(x) + 1} \right) \]</span></p>
<p>where <span class="math inline">\(\eta(x)\)</span> is the height of a term and <span class="math inline">\(\delta(x)\)</span> is the depth of the term. 1 is added in the formula to get rid of <span class="math inline">\(\log(0)\)</span> for leaf terms.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/term_IC.html">term_IC</a></span><span class="op">(</span><span class="va">dag</span>, method <span class="op">=</span> <span class="st">"IC_height"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="ic_annotation">IC_annotation<a class="anchor" aria-label="anchor" href="#ic_annotation"></a>
</h2>
<p>Due to the nature of DAG, if a child term is annotated to an item, all its ancestor terms are also associated with that item. The calculation of annotated items is applied in a recursive way.</p>
<p>Denote <span class="math inline">\(G^*_x\)</span> as the set of items <em>directly</em> annotated to <span class="math inline">\(x\)</span>, and <span class="math inline">\(G_x\)</span> as the set of items annotated to <span class="math inline">\(x\)</span> after merging from its child terms. <span class="math inline">\(G_x\)</span> is the union of all items annotated to its child terms and the items directly annotated to <span class="math inline">\(x\)</span>.</p>
<p><span class="math display">\[ G_x = \left( \bigcup_{z \in \mathcal{C}_x} G_z \right) \bigcup G^*_x \]</span></p>
<p>And the information content is:</p>
<p><span class="math display">\[ \mathrm{IC}(x) = -\log \left( \frac{|G_x|}{|G_r|} \right) \]</span></p>
<p>where <span class="math inline">\(G_r\)</span> is the set of all items annotated to the DAG (i.e. to the root).</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/term_IC.html">term_IC</a></span><span class="op">(</span><span class="va">dag</span>, method <span class="op">=</span> <span class="st">"IC_annotation"</span><span class="op">)</span></span></code></pre></div>
<p><code>NA</code> is assigned to the terms with no item annotated.</p>
<p>In some implementation of current tools, number of annotated items on a term is calculated by simply adding such numbers from all its child terms, as shown in the following equation:</p>
<p><span class="math display">\[ f(x) = \left\{ \begin{array}{lr}
\beta(x) &amp; \text{if } \textit{x} \text{ is a leaf} \\
\beta(x) + \sum\limits_{z \in \mathcal{C}_x} \beta(z) &amp; \text{otherwise}
\end{array} \right. \]</span></p>
<p>where <span class="math inline">\(\beta(x)\)</span> is the number of items directly annotated to term <span class="math inline">\(x\)</span>. It should be noted that this way over-estimates the number of annotated items because an item can be annotated to multiple terms. For example, a gene can be annotated to multiple GO terms (you can check the object <code><a href="https://rdrr.io/pkg/org.Hs.eg.db/man/org.Hs.egGO.html" class="external-link">org.Hs.eg.db::org.Hs.egGO</a></code>. And even for the same term, it may have duplicated annotated genes with multiple evidence codes (e.g. the gene with Entrez ID <code>"994"</code> is annotated to a GO term <code>"GO:0000086"</code> twice with evidence code <code>"IBA"</code> and <code>"TAS"</code>).</p>
<p>Since annotated-based <span class="math inline">\(p(x)\)</span> is the probability of a term being annotated when randomly selecting an item, the second way might not be proper. The two methods are identical only if an item is uniquely annotated to a term in the DAG.</p>
<p>We can compare the two different implementations with or without uniquifying annotations. We can specify an additional parameter for the <em>IC_annotation</em> method: <code>control = list(uniquify = FALSE)</code> to use the second method.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ic1</span> <span class="op">=</span> <span class="fu"><a href="../reference/term_IC.html">term_IC</a></span><span class="op">(</span><span class="va">dag</span>, method <span class="op">=</span> <span class="st">"IC_annotation"</span><span class="op">)</span></span>
<span><span class="va">ic2</span> <span class="op">=</span> <span class="fu"><a href="../reference/term_IC.html">term_IC</a></span><span class="op">(</span><span class="va">dag</span>, method <span class="op">=</span> <span class="st">"IC_annotation"</span>, control <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>uniquify <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># ranges on both x- and y-axes</span></span>
<span><span class="va">rg</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">ic1</span>, <span class="va">ic2</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">ic1</span>, <span class="va">ic2</span>, xlim <span class="op">=</span> <span class="va">rg</span>, ylim <span class="op">=</span> <span class="va">rg</span>,</span>
<span>    xlab <span class="op">=</span> <span class="st">"uniquified (first method)"</span>, ylab <span class="op">=</span> <span class="st">"not uniquified (second method)"</span>, </span>
<span>    pch <span class="op">=</span> <span class="fl">16</span>, col <span class="op">=</span> <span class="st">"#00000020"</span>, main <span class="op">=</span> <span class="st">"compare IC_annotation"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/abline.html" class="external-link">abline</a></span><span class="op">(</span>a <span class="op">=</span> <span class="fl">0</span>, b <span class="op">=</span> <span class="fl">1</span>, col <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span></span></code></pre></div>
<p><img src="v4_information_content_files/figure-html/unnamed-chunk-4-1.png" width="576" style="display: block; margin: auto;"></p>
<p>It shows there is a systematic shift between the two methods.</p>
<p><em>IC_annotation</em> is the only method which replies on external annotation data. This type of method has been criticized because it replies on changable external data and people argue that IC should be an intrinsic attribute and only depends on the DAG structure itself. All other IC methods introduced in later sections only purely reply on the DAG structure itself.</p>
</div>
<div class="section level2">
<h2 id="ic_universal">IC_universal<a class="anchor" aria-label="anchor" href="#ic_universal"></a>
</h2>
<p>It measures the probability of a term getting full transmission from the root. Each term is associated with a p-value and the root has the p-value of 1.</p>
<p>To understand this method, let’s think the DAG relates to a kind of information transmission from the root to the inner DAG, where information is 100% on root. Then, in a certain level in the DAG, a term <span class="math inline">\(x\)</span> has two parents <span class="math inline">\(t_1\)</span> and <span class="math inline">\(t_2\)</span>, also let’s assume <span class="math inline">\(t_1\)</span> has <span class="math inline">\(k_1\)</span> children and <span class="math inline">\(t_2\)</span> has <span class="math inline">\(k_2\)</span> children, next we assume a parent transmits information equally to all its children, then <span class="math inline">\(t_1\)</span> only transmits <span class="math inline">\(1/k_1\)</span> and <span class="math inline">\(t_2\)</span> only transmits <span class="math inline">\(1/k_2\)</span> of its content to term <span class="math inline">\(x\)</span>, on which we could also say, the probability of <span class="math inline">\(t_1\)</span> or <span class="math inline">\(t_2\)</span> to reach <span class="math inline">\(x\)</span> is <span class="math inline">\(1/k_1\)</span> or <span class="math inline">\(1/k_2\)</span>. Let’s assume <span class="math inline">\(p(t_1)\)</span> and <span class="math inline">\(p(t_2)\)</span> are the accumulated contents from the root to <span class="math inline">\(t_1\)</span> and <span class="math inline">\(t_2\)</span> already (or the probability of the two parent terms getting full transmission from root), then the probability of reaching <span class="math inline">\(x\)</span> via a full transmission graph from <span class="math inline">\(t_1\)</span> is the multiplication of <span class="math inline">\(p(t_1)\)</span> and <span class="math inline">\(1/k_1\)</span>, which is <span class="math inline">\(p(t_1)/k_1\)</span>, and it is similar for <span class="math inline">\(p(t_2)/k_2\)</span>. Then, for term <span class="math inline">\(x\)</span>, if getting transmitted from <span class="math inline">\(t_1\)</span> and <span class="math inline">\(t_2\)</span> are independent, the probability of <span class="math inline">\(x\)</span> (denoted as <span class="math inline">\(p(x)\)</span>) to get transmitted from both parents is:</p>
<p><span class="math display">\[ p(x) = \frac{p(t_1)}{k_1} * \frac{p(t_2)}{k_2} \]</span></p>
<p>Since the two parents are the full set of <span class="math inline">\(x\)</span>’s parents, <span class="math inline">\(p(x)\)</span> is the probability of <span class="math inline">\(x\)</span> getting full transmission from root. And the final information content is:</p>
<p><span class="math display">\[ \mathrm{IC}(x) = -\log(p(x)) \]</span></p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/term_IC.html">term_IC</a></span><span class="op">(</span><span class="va">dag</span>, method <span class="op">=</span> <span class="st">"IC_universal"</span><span class="op">)</span></span></code></pre></div>
<p>The formal definition is:</p>
<p><span class="math display">\[ p(x) = \left\{ \begin{array}{cl}
1 &amp; \text{if } \textit{x} \text{ is a root} \\
\prod\limits_{t \in \mathcal{P}_x} \frac{p(t)}{|\mathcal{C}_t|} &amp; \text{otherwise}
\end{array} \right. \]</span></p>
<p>Paper link: <a href="https://doi.org/10.1155/2012/975783" class="external-link uri">https://doi.org/10.1155/2012/975783</a>.</p>
</div>
<div class="section level2">
<h2 id="ic_zhang_2006">IC_Zhang_2006<a class="anchor" aria-label="anchor" href="#ic_zhang_2006"></a>
</h2>
<p>It measures the number of ways of a term to reach leaf terms. E.g. in the following DAG:</p>
<pre><code>     a      upstream
    /|\
   b | c
     |/
     d      downstream</code></pre>
<p>Term “a” has three ways to reach leaves, which are <code>a-&gt;b</code>, <code>a-&gt;d</code> and <code>a-&gt;c-&gt;d</code>.</p>
<p>The number of ways of term <span class="math inline">\(x\)</span> to reach leaf terms can be obtained in a recursively way:</p>
<p><span class="math display">\[ f(x) = \left\{ \begin{array}{cl}
1 &amp; \text{if } \textit{x} \text{ is a leaf} \\
\sum\limits_{t \in \mathcal{C}_x} f(t) &amp; \text{otherwise}
\end{array} \right. \]</span></p>
<p>And the information content:</p>
<p><span class="math display">\[ \mathrm{IC}(x) = -\log \left( \frac{f(x)}{f(r)} \right) \]</span></p>
<p>where <span class="math inline">\(f(r)\)</span> is the number of ways to reach all leaf terms from root.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/term_IC.html">term_IC</a></span><span class="op">(</span><span class="va">dag</span>, method <span class="op">=</span> <span class="st">"IC_Zhang_2006"</span><span class="op">)</span></span></code></pre></div>
<p>Paper link: <a href="https://doi.org/10.1186/1471-2105-7-135" class="external-link uri">https://doi.org/10.1186/1471-2105-7-135</a>.</p>
</div>
<div class="section level2">
<h2 id="ic_seco_2004">IC_Seco_2004<a class="anchor" aria-label="anchor" href="#ic_seco_2004"></a>
</h2>
<p>It is based on the number of offspring terms (with itself) of term <span class="math inline">\(x\)</span>. The information content is calculated as:</p>
<p><span class="math display">\[ 
\begin{align*}
\mathrm{IC}(x) &amp; = \frac{\log(\frac{|\mathcal{D}_x^+|}{|\mathcal{D}_r^+|})}{\log(\frac{1}{|\mathcal{D}_r^+|})}  \\
 &amp; = 1 - \frac{\log(|\mathcal{D}_x^+|)}{\log(|\mathcal{D}_r^+|)}
\end{align*} 
\]</span></p>
<p>where <span class="math inline">\(\|\mathcal{D}_r^+\|\)</span> is the total number of terms in the DAG.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/term_IC.html">term_IC</a></span><span class="op">(</span><span class="va">dag</span>, method <span class="op">=</span> <span class="st">"IC_Seco_2004"</span><span class="op">)</span></span></code></pre></div>
<p><em>IC_Seco_2004</em> is actually a normalized version of <em>IC_offspring</em>:</p>
<p><span class="math display">\[
\begin{align*}
\mathrm{IC}_\mathrm{Seco}(x) &amp; = 1 - \frac{\log(|\mathcal{D}_x^+|)}{\log(|\mathcal{D}_r^+|)} \\
&amp; = \frac{\log(|\mathcal{D}_r^+|) - \log(|\mathcal{D}_x^+|)}{\log(|\mathcal{D}_r^+|)} \\
&amp; = \frac{-\log (|\mathcal{D}_x^+| / |\mathcal{D}_r^+|)}{\log(|\mathcal{D}_r^+|)} \\
&amp; = \frac{\mathrm{IC}_\mathrm{offspring}(x)}{\log(|\mathcal{D}_r^+|)}
\end{align*}
\]</span></p>
<p>Paper link: <a href="https://dl.acm.org/doi/10.5555/3000001.3000272" class="external-link uri">https://dl.acm.org/doi/10.5555/3000001.3000272</a>.</p>
</div>
<div class="section level2">
<h2 id="ic_zhou_2008">IC_Zhou_2008<a class="anchor" aria-label="anchor" href="#ic_zhou_2008"></a>
</h2>
<p>It is a correction of <em>IC_Seco_2004</em> which also considers the depth of a term in the DAG. The information content is calculated as:</p>
<p><span class="math display">\[ \mathrm{IC}(x) = 0.5 * \mathrm{IC}_\mathrm{Seco}(x) + 0.5*\frac{\log(\delta(x))}{\log(\delta_\mathrm{max})} \]</span></p>
<p>And IC is set to 0 for root term.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/term_IC.html">term_IC</a></span><span class="op">(</span><span class="va">dag</span>, method <span class="op">=</span> <span class="st">"IC_Zhou_2008"</span><span class="op">)</span></span></code></pre></div>
<p>Paper link: <a href="https://doi.org/10.1109/FGCNS.2008.16" class="external-link uri">https://doi.org/10.1109/FGCNS.2008.16</a>.</p>
</div>
<div class="section level2">
<h2 id="ic_seddiqui_2010">IC_Seddiqui_2010<a class="anchor" aria-label="anchor" href="#ic_seddiqui_2010"></a>
</h2>
<p>It is also a correction of <em>IC_Seco_2004</em> but considers the number of relations a term has in the DAG.</p>
<p><span class="math display">\[ \mathrm{IC}(x) = (1 - \sigma) * \mathrm{IC}_\mathrm{Seco}(x) + \sigma*\frac{\log(\lambda(x) + 1)}{\log(\lambda_\mathrm{edge} + 1)} \]</span></p>
<p>where <span class="math inline">\(\lambda(x)\)</span> is the number of relations of <span class="math inline">\(x\)</span>, i.e. number of its parent and child terms, <span class="math inline">\(\lambda_\mathrm{edge}\)</span> is the total number of edges in the DAG. The tunning factor <span class="math inline">\(\sigma\)</span> is defined as:</p>
<p><span class="math display">\[ \sigma = \frac{\log(\lambda_\mathrm{edge}+1)}{\log(\lambda_\mathrm{edge}) + \log(\lambda_\mathrm{node})} \]</span></p>
<p>where <span class="math inline">\(\lambda_\mathrm{node}\)</span> is the total number of nodes in the DAG. When the DAG has a tree structure, i.e. a term only has one parent and root has no parent, there is the following relation:</p>
<p><span class="math display">\[ \lambda_\mathrm{node} = \lambda_\mathrm{edge} + 1\]</span></p>
<p>In this case, <span class="math inline">\(\sigma \approx 0.5\)</span>.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/term_IC.html">term_IC</a></span><span class="op">(</span><span class="va">dag</span>, method <span class="op">=</span> <span class="st">"IC_Seddiqui_2010"</span><span class="op">)</span></span></code></pre></div>
<p>Paper link: <a href="https://dl.acm.org/doi/10.5555/1862330.1862343" class="external-link uri">https://dl.acm.org/doi/10.5555/1862330.1862343</a>.</p>
</div>
<div class="section level2">
<h2 id="ic_sanchez_2011">IC_Sanchez_2011<a class="anchor" aria-label="anchor" href="#ic_sanchez_2011"></a>
</h2>
<p>It measures the average contribution of term <span class="math inline">\(x\)</span> on leaf terms. First denote <span class="math inline">\(\zeta_x\)</span> as the number of leaf terms that can be reached from term <span class="math inline">\(x\)</span>. Since all <span class="math inline">\(x\)</span>’s ancestors can also reach <span class="math inline">\(x\)</span>’s leaves, the contribution of <span class="math inline">\(x\)</span> on leaf terms is scaled by the number of its ancestors. The final information content is normalized by the total number of leaves in the DAG.</p>
<p><span class="math display">\[ \mathrm{IC}(x) = -\log \left( \frac{\zeta_x}{|\mathcal{A}_x^+| * \zeta_\mathrm{leaf}} \right) \]</span></p>
<p>where <span class="math inline">\(\zeta_\mathrm{max}\)</span> is the total number of leaves in the DAG.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/term_IC.html">term_IC</a></span><span class="op">(</span><span class="va">dag</span>, method <span class="op">=</span> <span class="st">"IC_Sanchez_2011"</span><span class="op">)</span></span></code></pre></div>
<p>Paper link: <a href="https://doi.org/10.1016/j.knosys.2010.10.001" class="external-link uri">https://doi.org/10.1016/j.knosys.2010.10.001</a>.</p>
</div>
<div class="section level2">
<h2 id="ic_meng_2012">IC_Meng_2012<a class="anchor" aria-label="anchor" href="#ic_meng_2012"></a>
</h2>
<p>It has a complex form which takes account of the term depth and the downstream of the term. The first factor <span class="math inline">\(r_1\)</span> is calculated as:</p>
<p><span class="math display">\[ r_1 = \frac{\log(\delta(x))}{\log(\delta_\mathrm{max})} \]</span></p>
<p>The second factor <span class="math inline">\(r_2\)</span> is calculated as:</p>
<p><span class="math display">\[ r_2 = 1 - \frac{\log \left(1 + \sum\limits_{t \in \mathcal{D}_x} \frac{1}{\delta(t)} \right)}{\log(\lambda_\mathrm{node})} \]</span></p>
<p>In the equation, the summation goes over <span class="math inline">\(t\)</span>’s offspring terms.</p>
<p>The final information content is the multiplication of <span class="math inline">\(r_1\)</span> and <span class="math inline">\(r_2\)</span>:</p>
<p><span class="math display">\[ \mathrm{IC}(x) = r_1 * r_2 \]</span></p>
<p>IC for root is set to 0.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/term_IC.html">term_IC</a></span><span class="op">(</span><span class="va">dag</span>, method <span class="op">=</span> <span class="st">"IC_Meng_2012"</span><span class="op">)</span></span></code></pre></div>
<p>Paper link: <a href="http://article.nadiapub.com/IJGDC/vol5_no3/6.pdf" class="external-link uri">http://article.nadiapub.com/IJGDC/vol5_no3/6.pdf</a>.</p>
</div>
<div class="section level2">
<h2 id="ic_wang_2007">IC_Wang_2007<a class="anchor" aria-label="anchor" href="#ic_wang_2007"></a>
</h2>
<p>Each child-parent relation is weighted by a value smaller than 1 based on the semantic relation type, i.e. 0.8 for “is_a” and 0.6 for “part_of”. For a term <span class="math inline">\(x\)</span> and one of its ancestor term <span class="math inline">\(t\)</span>, it first calculates an “S-value” which corresponds to a path from <span class="math inline">\(t\)</span> to <span class="math inline">\(x\)</span> where the accumulated multiplication of weights along the path reaches maximal:</p>
<p><span class="math display">\[ S_x(t) = \left\{ \begin{array}{cl}
1 &amp; \text{if } \textit{t = x} \\
\max\limits_{t' \in \mathcal{C}_t}{w_e * S_x(t')} &amp; \text{otherwise}
\end{array} \right. \]</span></p>
<p>To properly understand this setting, let’s first apply the following transformation:</p>
<p><span class="math display">\[ T_x(t) = \log(1/S_x(t)) \]</span></p>
<p>And the definition from <span class="math inline">\(S_x(t)\)</span> is converted to <span class="math inline">\(T_x(t)\)</span>:</p>
<p><span class="math display">\[ T_x(t) = \left\{ \begin{array}{cl}
0 &amp; \text{if } \textit{t = x} \\
\min\limits_{t' \in \mathcal{C}_t}{\frac{1}{w_e} * T_x(t')} &amp; \text{otherwise}
\end{array} \right. \]</span></p>
<p>In this way, <span class="math inline">\(T_x(t)\)</span> is actually the shortest distance from <span class="math inline">\(t\)</span> to <span class="math inline">\(x\)</span> taking <span class="math inline">\(1/w_e\)</span> as the edge weight. We can obtain <span class="math inline">\(S_x(t)\)</span> back by:</p>
<p><span class="math display">\[ S_x(t) = \exp(-T_x(t)) \]</span></p>
<p>The information content is the summation over all <span class="math inline">\(x\)</span>’s ancestors:</p>
<p><span class="math display">\[ \mathrm{IC}(x) = \sum\limits_{t \in \mathcal{A}_x^+} S_x(t) \]</span></p>
<p>Normally <em>IC_Wang_2007</em> is only used for calculating the semantic similarity using <em>Sim_Wang_2007</em> method.</p>
<p>The contribution of different semantic relations can be set with the <code>contribution_factor</code> parameter. The value should be a named numeric vector where names should cover the relations defined in relations set in <code><a href="../reference/create_ontology_DAG.html">create_ontology_DAG()</a></code>. For example, if there are two relations “relation_a” and “relation_b” set in the DAG, the value for contribution_factor can be set as:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/term_IC.html">term_IC</a></span><span class="op">(</span><span class="va">dag</span>, method <span class="op">=</span> <span class="st">"IC_Wang_2007"</span>, </span>
<span>    control <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>contribution_factor <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"relation_a"</span> <span class="op">=</span> <span class="fl">0.8</span>, <span class="st">"relation_b"</span> <span class="op">=</span> <span class="fl">0.6</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>By default 0.8 is set for “is_a” and 0.6 for “part_of”.</p>
<p>If you are not sure what types of relations have been set, simply type the <code>dag</code> object. The relation types are all printed there.</p>
<p>Paper link: <a href="https://doi.org/10.1093/bioinformatics/btm087" class="external-link uri">https://doi.org/10.1093/bioinformatics/btm087</a>.</p>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.2.0 (2022-04-22)</span></span>
<span><span class="co">## Platform: x86_64-apple-darwin17.0 (64-bit)</span></span>
<span><span class="co">## Running under: macOS Big Sur/Monterey 10.16</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib</span></span>
<span><span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] C/UTF-8/C/C/C/C</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] simona_0.99.0 knitr_1.43   </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] Rcpp_1.0.11            circlize_0.4.16        GO.db_3.16.0          </span></span>
<span><span class="co">##  [4] Biostrings_2.66.0      png_0.1-8              rprojroot_2.0.3       </span></span>
<span><span class="co">##  [7] digest_0.6.33          foreach_1.5.2          GenomeInfoDb_1.34.9   </span></span>
<span><span class="co">## [10] R6_2.5.1               stats4_4.2.0           Polychrome_1.5.1      </span></span>
<span><span class="co">## [13] RSQLite_2.3.1          evaluate_0.21          highr_0.10            </span></span>
<span><span class="co">## [16] httr_1.4.6             zlibbioc_1.44.0        GlobalOptions_0.1.2   </span></span>
<span><span class="co">## [19] rlang_1.1.1            jquerylib_0.1.4        blob_1.2.4            </span></span>
<span><span class="co">## [22] S4Vectors_0.36.2       GetoptLong_1.1.0       rmarkdown_2.23        </span></span>
<span><span class="co">## [25] pkgdown_2.0.7          textshaping_0.3.6      desc_1.4.2            </span></span>
<span><span class="co">## [28] stringr_1.5.0          RCurl_1.98-1.12        igraph_1.5.0          </span></span>
<span><span class="co">## [31] bit_4.0.5              compiler_4.2.0         xfun_0.39             </span></span>
<span><span class="co">## [34] pkgconfig_2.0.3        systemfonts_1.0.4      BiocGenerics_0.44.0   </span></span>
<span><span class="co">## [37] shape_1.4.6            htmltools_0.5.5        KEGGREST_1.38.0       </span></span>
<span><span class="co">## [40] GenomeInfoDbData_1.2.9 IRanges_2.32.0         codetools_0.2-19      </span></span>
<span><span class="co">## [43] matrixStats_1.0.0      crayon_1.5.2           bitops_1.0-7          </span></span>
<span><span class="co">## [46] grid_4.2.0             jsonlite_1.8.7         lifecycle_1.0.3       </span></span>
<span><span class="co">## [49] DBI_1.1.3              magrittr_2.0.3         cli_3.6.1             </span></span>
<span><span class="co">## [52] stringi_1.7.12         cachem_1.0.8           XVector_0.38.0        </span></span>
<span><span class="co">## [55] fs_1.6.2               scatterplot3d_0.3-44   doParallel_1.0.17     </span></span>
<span><span class="co">## [58] xml2_1.3.5             bslib_0.5.0            ragg_1.2.5            </span></span>
<span><span class="co">## [61] vctrs_0.6.3            org.Hs.eg.db_3.16.0    rjson_0.2.21          </span></span>
<span><span class="co">## [64] RColorBrewer_1.1-3     iterators_1.0.14       tools_4.2.0           </span></span>
<span><span class="co">## [67] bit64_4.0.5            Biobase_2.58.0         glue_1.6.2            </span></span>
<span><span class="co">## [70] purrr_1.0.1            parallel_4.2.0         fastmap_1.1.1         </span></span>
<span><span class="co">## [73] yaml_2.3.7             clue_0.3-64            AnnotationDbi_1.60.2  </span></span>
<span><span class="co">## [76] colorspace_2.1-0       cluster_2.1.4          TSP_1.2-4             </span></span>
<span><span class="co">## [79] ComplexHeatmap_2.15.4  memoise_2.0.1          sass_0.4.7</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Zuguang Gu.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
